### PR TITLE
Parse phone number mentions on mobile

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

* On mobile these are rendered as links, on desktop as text.
* Only works for US formats like `(123) 456-7890`, `123-456-7890`, `123 456 7890`, see the tests for a complete list.
* For that last one (`123 456 7890`) there was a problem in that we weren't sending text to the markdown parser if it didn't contain any special chars.  So I added an explicit test for phone numbers to that test.

### CODE REVIEW

File: shared/markdown/parser.pegjs [hunk 1]
```diff
@@ -118,6 +120,14 @@ Channel
     options && options.channelNameToConvID && options.channelNameToConvID(name)
 } { return {type: 'channel', children: [name], convID: options && options.channelNameToConvID && options.channelNameToConvID(name) } }
 
+Num = [0-9]
+Phone
+ = phone:($ (
+   (PhoneMarker? Num Num Num PhonePostfix Num Num Num [- ] Num Num Num Num !NonBlank) /
+   (PhoneMarker? Num Num Num [- ] Num Num Num [- ] Num Num Num Num !NonBlank))) {
```
Comment (buoyad) [inline shared/markdown/parser.pegjs]: Would it cause problems to add a variant `Num{10}Num?` to support phone numbers with no formatting?

File: shared/markdown/parser.pegjs [hunk 2]
```diff
@@ -220,13 +230,10 @@ Link
...
     const matches = linkExp.exec(URL)
-    url._data = {leading: leading, matches: matches, trailing: trailing, URL: URL}
+    url._data = {leading, matches, trailing, URL}
     return matches
   } {
-    const leading = url._data.leading
-    const matches = url._data.matches
-    const trailing = url._data.trailing
-    const URL = url._data.URL
+    const {leading, matches, trailing, URL} = url._data
```
Comment (buoyad) [inline shared/markdown/parser.pegjs]: 💯

File: shared/markdown/parser.pegjs (hunk 1 above)
Comment (cjb) [reply shared/markdown/parser.pegjs]: @buoyad Makes sense, done, PTAL!

**Comment by cjb** [comment]
@zanderz Thanks, added your suggested syntax too!

**Pull request merged**